src/ghosting/non_manifold_coupling.C

```diff
@@ -0,0 +1,101 @@
+// The libMesh Finite Element Library.
+// Copyright (C) 2002-2024 Benjamin S. Kirk, John W. Peterson, Roy H. Stogner
+
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public  License for more details.
+
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+
+// libMesh includes
+#include "libmesh/non_manifold_coupling.h"
+#include "libmesh/elem.h"
+
+namespace libMesh
+{
+
+NonManifoldGhostingFunctor::
+NonManifoldGhostingFunctor(const MeshBase & mesh) :
+  GhostingFunctor(mesh),
+  _stem(MeshTools::SidesToElemMap::build(mesh))
+{
+  // TODO: find out if we should build the cache here in the constructor
+  // or if there will always be at least one mesh_reinit() call before the
+  // NonManifoldGhostingFunctor is called. Roy's initial suggestion is to
+  // build the cache in both places.
```

Comment:
I checked; we do a `mesh_reinit()` call in `prepare_for_use()`, before we ever try to use ghosting functors for anything on the mesh or its systems.  If the ghosting functor is attached before the mesh is initialized, which it very much should be because otherwise we can end up deleting remote elements that can't be recovered without serializing, then skipping the map build here should be safe.

On the other hand, if someone's *really* doing things properly then they should be attaching the ghosting functor before ever putting elements in the mesh (in which case the build should be free to leave in), and if someone's doing things a bit dubiously (cough, MOOSE) then better safe than sorry.